Lib/cu2qu/rf.py

```diff
@@ -46,6 +44,103 @@ def zip(*args):
     return _zip(*args)
 
 
+class GetSegmentsPen(AbstractPen):
+    """Pen to collect segments into lists of points for conversion."""
+
+    def __init__(self):
+        self._last_pt = None
+        self.segments = []
+
+    def _add_segment(self, tag, *args):
+        self.segments.append((tag, args))
+
+    def moveTo(self, pt):
+        self._add_segment('move', pt)
+        self._last_pt = pt
+
+    def lineTo(self, pt):
+        self._add_segment('line', pt)
+        self._last_pt = pt
+
+    def curveTo(self, *points):
+        self._add_segment('curve', self._last_pt, *points)
```

Comment:
Store in the `_add_segment()` call I meant.

On Mon, Apr 4, 2016 at 4:51 PM, James Godfrey-Kittle <
notifications@github.com> wrote:

> In Lib/cu2qu/rf.py
> <https://github.com/googlei18n/cu2qu/pull/21#discussion_r58468273>:
>
> > +        self._last_pt = None
> > +        self.segments = []
> > +
> > +    def _add_segment(self, tag, *args):
> > +        self.segments.append((tag, args))
> > +
> > +    def moveTo(self, pt):
> > +        self._add_segment('move', pt)
> > +        self._last_pt = pt
> > +
> > +    def lineTo(self, pt):
> > +        self._add_segment('line', pt)
> > +        self._last_pt = pt
> > +
> > +    def curveTo(self, *points):
> > +        self._add_segment('curve', self._last_pt, *points)
>
> lineTo does store the last point, right? (line 63)
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/googlei18n/cu2qu/pull/21/files/7b816131f873cada90221dec82e6a044bbf3a34d#r58468273>
>



-- 
behdad
http://behdad.org/